pkg/health/checker/checker_test.go

```diff
@@ -24,6 +26,157 @@ func (f fakeConsulStore) GetServiceHealth(service string) (map[string]consul.Wat
 	return f.results, nil
 }
 
+type fakeHealthClient struct {
+	HealthResponses map[string]hc.HealthResponse
+	MonitorDelay    time.Duration
+}
+
+func (f fakeHealthClient) HealthCheck(ctx context.Context, req *hc.HealthRequest) (health.HealthState, error) {
+	return f.HealthResponses[req.Url].Health, nil
+}
+
+func (f fakeHealthClient) HealthMonitor(ctx context.Context, req *hc.HealthRequest, resultCh chan *hc.HealthResponse) error {
+	timer := time.NewTimer(0)
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-timer.C:
+			timer.Reset(f.MonitorDelay)
```

Comment:
changing this to a smaller time (like 200 millis) makes the test run faster and it still passes. Is that a safe change to make?

Reply:
- It shouldn't make the test run faster because initial messages are sent right away, the initial timer is set to 0
- this always passes when there isn't a regression
- when there is a regression and the oldResults are received is when this test will fail

Reply:
Fair enough. The test sometimes takes 5s to pass on my machine, other times it takes 0.3s. It seems independent of my random changes :)

Reply:
Any idea why this test sometimes takes 5s to run?

Reply:
It's consistently 5s for me. Probably just the timing of when the oldResults are sent.